EgoLanes/create_lane/BDD100K/process_bdd100k.py

```diff
@@ -0,0 +1,806 @@
+    )
+
+
+if __name__ == "__main__":
+    # ============================== Constants ============================== #
+
+    # Define Dimension
+    ORIGINAL_IMG_WIDTH = 1280
+    ORIGINAL_IMG_HEIGHT = 720
+
+    # ============================== Parsing args ============================== #
+
+    parser = argparse.ArgumentParser(
+        description="Process BDD100k dataset - EgoLanes groundtruth generation"
+    )
+
+    # bdd100k/images/100k/
+    parser.add_argument(
+        "--image_dir",
+        type=str,
+        help="BDD100k ground truth image parent directory (right after extraction)",
+    )
+
+    # bdd100k/labels/
+    parser.add_argument(
+        "--labels_dir",
+        type=str,
+        help="BDD100k labels directory (right after extraction)",
+        required=True,
+    )
+
+    # ./output
+    parser.add_argument(
+        "--output_dir", type=str, help="Desired output directory", required=True
+    )
+
+    parser.add_argument(
+        "--crop",
+        type=int,
+        nargs=4,
+        help="Crop image: [TOP, RIGHT, BOTTOM, LEFT]. Must always be 4 ints. Non-cropped sizes are 0.",
+        metavar=("TOP", "RIGHT", "BOTTOM", "LEFT"),
+        default=[0, 140, 220, 140],
+        required=False,
+    )
+
+    args = parser.parse_args()
+
+    json_data_path = os.path.join(
+        args.labels_dir, "lane", "polygons", "lane_train.json"
+    )
+
+    # Save ground truth images.
+    gt_images_path = os.path.join(args.image_dir, "train")
+    data = saveGT(
+        json_data_path, gt_images_path, args
+    )  # Do not enable if not required.
+
+    # ============================== Get crop image params ============================== #
+
+    # Updated dimensions
+    IMG_WIDTH = ORIGINAL_IMG_WIDTH - args.crop[1] - args.crop[3]
+    IMG_HEIGHT = ORIGINAL_IMG_HEIGHT - args.crop[0] - args.crop[2]
+
+    crop = {
+        "TOP": args.crop[0],
+        "RIGHT": args.crop[1],
+        "BOTTOM": args.crop[2],
+        "LEFT": args.crop[3],
+    }
+
+    # ============================== Save binary mask ============================== #
+    process_binary_mask(args)
+
+    # ============================== Identify EgoLanes ============================== #
+    gt_images_path = os.path.join(args.output_dir, "images")
+    classified_lanes = classify_lanes(data, gt_images_path, output_dir=args.output_dir)
+
+    # # # ============================== Normalise, crop Image data  ============================== #
+    formatted_data = format_data(classified_lanes, crop)
+
+    # # # ============================== AnnotateGT ================================ # # #
+
+    annotateGT(formatted_data, gt_images_path, output_dir=args.output_dir, crop=crop)
+
+    # # # ============================== Save result JSON ============================== #
+
+    # Save classified lanes as new JSON file
+    output_file = os.path.join(args.output_dir, "bdd100k_egolanes_train.json")
```

Comment:
Can you change the name of output JSON file into `drivable_path.json` ? This is to make sure your processing pipeline follows our [standardized rule](https://github.com/autowarefoundation/autoware.privately-owned-vehicles/issues/64).